File: services/product-registry/api/client/src/main/java/demo/craft/user/profile/client/model/Product.java
package demo.craft.user.profile.client.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonValue;
import org.openapitools.jackson.nullable.JsonNullable;
import javax.validation.Valid;
import javax.validation.constraints.*;

import com.fasterxml.jackson.annotation.JsonCreator;

/**
 * Gets or Sets Product
 */
public enum Product {
  
  QUICKBOOKS_ACCOUNTING("QUICKBOOKS_ACCOUNTING"),
  
  QUICKBOOKS_PAYROLL("QUICKBOOKS_PAYROLL"),
  
  QUICKBOOKS_PAYMENTS("QUICKBOOKS_PAYMENTS"),
  
  TSHEETS("TSHEETS");

  private String value;

  Product(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static Product fromValue(String value) {
    for (Product b : Product.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
}

File: services/product-registry/api/server-stub/src/main/kotlin/demo/craft/product/registry/model/Product.kt
package demo.craft.product.registry.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonValue
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size

/**
* 
* Values: QUICKBOOKS_ACCOUNTING,QUICKBOOKS_PAYROLL,QUICKBOOKS_PAYMENTS,TSHEETS
*/
enum class Product(val value: kotlin.String) {

    QUICKBOOKS_ACCOUNTING("QUICKBOOKS_ACCOUNTING"),

    QUICKBOOKS_PAYROLL("QUICKBOOKS_PAYROLL"),

    QUICKBOOKS_PAYMENTS("QUICKBOOKS_PAYMENTS"),

    TSHEETS("TSHEETS");

}

